File: search-adapter-in/src/main/java/com/example/search/adapter/in/web/dto/AnalyticsDtos.java
package com.example.search.adapter.in.web.dto;

import java.util.List;

/**
 * 분석 admin REST DTO 모음. 도메인 record 와 분리해 응답 schema 안정성 확보.
 */
public final class AnalyticsDtos {

    private AnalyticsDtos() {
    }

    public record QueryStatDto(String keyword, long count) {
    }

    public record TopQueriesResponse(List<QueryStatDto> queries) {
    }

    public record LatencyResponse(long p50, long p95, long p99, long sampleSize) {
    }

    public record CtrResponse(long searchesWithResults, long clicks, double rate) {
    }
}

File: search-adapter-in/src/main/kotlin/com/example/search/adapter/in/web/dto/AnalyticsDtos.kt
package com.example.search.adapter.`in`.web.dto

/**
 * 분석 admin REST DTO 모음. 도메인 record 와 분리해 응답 schema 안정성 확보.
 */
object AnalyticsDtos {

    @JvmRecord
    data class QueryStatDto(val keyword: String, val count: Long)

    @JvmRecord
    data class TopQueriesResponse(val queries: List<QueryStatDto>)

    @JvmRecord
    data class LatencyResponse(val p50: Long, val p95: Long, val p99: Long, val sampleSize: Long)

    @JvmRecord
    data class CtrResponse(val searchesWithResults: Long, val clicks: Long, val rate: Double)
}
